### PR TITLE
Add Review schemas & connection functions

### PR DESCRIPTION
**Pull request by Vinuyans**
Added Schemas for:
- Reviews
- Assignment
- Group

Added functions to do the following actions on the DB:
- Add group to course
- Add assignment to group
- Add review to assignment
- Add user to group

### CODE REVIEW

**Comment by Vinuyans** [comment]
Relates to all issues regarding reviews and groups but does not close them

File: server/routes/api.js [hunk 1]
```diff
@@ -1,12 +1,36 @@
...
+  const group = await db.addGroupToCourse(course._id, ["12345678"]);
+  db.addUserToCourseGroup("12345678", group._id);
+  const assignment = await db.addAssignmentToGroup(
+    group._id,
+    "bad content",
+    "12345678"
+  );
+  db.addReviewToGroup(group._id, "12345678", assignment._id, 2);
+  res.status(200).json({ message: "done" });
+});
```
Comment (Marian-Hristov) [inline server/routes/api.js]: if you are leaving this as showcase, please comment it out so it doesn't get used!

**Review by Marian-Hristov — COMMENTED** [review]
Great work! Just add `strict: true` to the Schema options for example:
`  { collections: "Groups", queryString: true }` -> `  { collections: "Groups", queryString: true, strict: true }`

**Comment by Vinuyans** [comment]
Yup that makes sense would you mind adding it rn. I'm currently unavailable to do this

File: server/routes/api.js [hunk 2]
```diff
@@ -1,12 +1,36 @@
...
+
+export default router;
```
Comment (Marian-Hristov) [inline server/routes/api.js]:
```suggestion
/* router.get("/test", async (req, res) => {
  db.addUser("Stue", "Dent", "student", "12345678", "Password!1");
  db.addUser("Dent", "Stue", "student", "12345679", "Password!2");
  db.addUser("Tech", "Art", "teacher", "87654321", "Password!1");
  const course = await db.addCourse(
    "311",
    "ENGR",
    "87654321",
    ["12345678"],
    []
  );
  const group = await db.addGroupToCourse(course._id, ["12345678"]);
  db.addUserToCourseGroup("12345678", group._id);
  const assignment = await db.addAssignmentToGroup(
    group._id,
    "bad content",
    "12345678"
  );
  db.addReviewToGroup(group._id, "12345678", assignment._id, 2);
  res.status(200).json({ message: "done" });
}); */

export default router;
```

File: server/db/schemas/assignment.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import mongoose, { Schema } from "mongoose";
+
+const assignmentSchema = new Schema(
+  {
+    content: String,
+    student_id: String,
+  },
+  { collections: "Assignments", queryString: true }
```
Comment (Marian-Hristov) [inline server/db/schemas/assignment.js]:
```suggestion
  { collections: "Assignments", queryString: true, strict: true }

```

File: server/db/schemas/group.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import mongoose, { Schema } from "mongoose";
+
+const groupSchema = new Schema(
+  {
+    course_id: String,
+    student_ids: [String],
+    review_ids: [String],
+    assignment_ids: [String],
+  },
+  { collections: "Groups", queryString: true }
```
Comment (Marian-Hristov) [inline server/db/schemas/group.js]:
```suggestion
  { collections: "Groups", queryString: true, strict: true }

```

File: server/db/schemas/review.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import mongoose, { Schema } from "mongoose";
+
+const reviewSchema = new Schema(
+  {
+    reviewer_id: String,
+    assignment_id: String,
+    rating: Number,
+  },
+  { collections: "Reviews", queryString: true }
```
Comment (Marian-Hristov) [inline server/db/schemas/review.js]:
```suggestion
  { collections: "Reviews", queryString: true, strict: true }

```

File: server/db/schemas/course.js [hunk 6]
```diff
@@ -2,12 +2,11 @@ import mongoose, { Schema } from "mongoose";
...
   {
-    course_id: String,
     number: String,
     dept: String,
     prof_id: String,
     student_ids: [String],
-    groups: [Array],
+    group_ids: [String],
   },
   { collections: "Courses", queryString: true }
```
Comment (Marian-Hristov) [inline server/db/schemas/course.js]:
```suggestion
  { collections: "Courses", queryString: true, strict: true }
```

**Review by Marian-Hristov — APPROVED** [review]
Approving changes above

**Pull request merged**